whole-tale/ngx-dashboard
# feat: connect / disconnect account modals

## Problem
During the Angular rewrite, apparently I forgot to re-implement the connect/disconnect account modals.

Fixes #82 

## Approach
Implement the functionality of these two modals, and fix the text/styling to match what we expect to see.

<img width="922" alt="Screen Shot 2020-08-17 at 2 08 23 PM" src="https://user-images.githubusercontent.com/1413653/90434512-27287a80-e093-11ea-8e8e-d2665f4592a4.png">

<img width="914" alt="Screen Shot 2020-08-17 at 2 04 59 PM" src="https://user-images.githubusercontent.com/1413653/90434400-f47e8200-e092-11ea-9f8a-0daa235524d0.png">
 
<img width="516" alt="Screen Shot 2020-08-17 at 2 05 02 PM" src="https://user-images.githubusercontent.com/1413653/90434396-f3e5eb80-e092-11ea-9179-2442c2edd4c2.png">

## How to Test
1. Checkout this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
3. Navigate to the User Settings view
    * You should see 3 providers listed here: Zenodo, DataONE, and Dataverse
    * You should see a "Connect Account" button beside each provider
4. Click the "Connect Account" button beside Dataverse or Zenodo
    * You should see the "Connect Account" modal display
    * You should see the resource servers / targets listed in the dropdown on the left
    * You should see that the "Connect" button is disabled without entering both a server choice and an API key
    * You should see a link to the chosen platform
5. Choose a resource server in the dropdown and enter a valid API key from that server
    * You should see the "Connect" button is now enabled
6. Click "Connect" at the bottom of the modal
    * You should see the modal dismiss
    * You should see a new list item for the chosen resource server appear beneath the provider you selected
    * You should see "Disconnect" beside each connected account / server
7. Repeat for as many providers as possible
    * You should see the same behavior for any Zenodo / Dataverse reource_server
8. Beside one of your connected servers, click "Disconnect"
    * You should see the "Disconnect Account" modal display asking for confirmation
9. Click "Confirm" at the bottom of the modal
    * You should see the modal dismiss
    * You should see the list item for the chosen resource server disappear
